File: android/src/newarch/java/com/oblador/storereview/StoreReviewModule.java
package com.oblador.storereview;

import androidx.annotation.NonNull;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import java.util.Map;
import java.util.HashMap;

public class StoreReviewModule extends NativeRNStoreReviewSpec {

    StoreReviewModule(ReactApplicationContext context) {
        super(context);
    }

    @Override
    @NonNull
    public String getName() {
        return StoreReviewModuleImpl.NAME;
    }

    @Override
    public void requestReview() {
        StoreReviewModuleImpl.requestReview(getReactApplicationContext());
    }
}

File: android/src/main/java/com/oblador/storereview/StoreReviewModule.kt
package com.oblador.storereview

import android.os.Build.VERSION.SDK_INT
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.google.android.play.core.review.ReviewManagerFactory

class StoreReviewModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "RNStoreReview"
    }

    override fun getConstants(): MutableMap<String, Any> {
        return mutableMapOf<String, Any>("isAvailable" to (SDK_INT >= 21))
    }

    @ReactMethod
    fun requestReview() {
        val manager = ReviewManagerFactory.create(reactApplicationContext)

        val requestReview = manager.requestReviewFlow()
        requestReview.addOnCompleteListener { request ->
            if (request.isSuccessful) {
                // We got the ReviewInfo object
                val reviewInfo = request.result
                val activity = currentActivity

                if (activity != null) {
                    val flow = manager.launchReviewFlow(activity, reviewInfo)
                    flow.addOnCompleteListener { _ ->
                        // The flow has finished. The API does not indicate whether the user
                        // reviewed or not, or even whether the review dialog was shown. Thus, no
                        // matter the result, we continue our app flow.
                    }


                }
            }
        }
    }

    
}
